src/manage_sql.c

```diff
@@ -56371,6 +56380,16 @@ hosts_set_details (report_t report)
        " AND (SELECT value = 'yes' FROM task_preferences"
        "      WHERE task = (SELECT task FROM reports WHERE id = %llu)"
        "      AND name = 'in_assets')"
+       /* Ensure that every report host detail has a corresponding host
+        *  in the assets. */
+       " AND EXISTS (SELECT host"
+       "               FROM host_identifiers"
+       "              WHERE source_id = (SELECT uuid FROM reports"
+       "                            WHERE id = %llu)"
+       "                AND (SELECT name FROM hosts WHERE id = host)"
+       "                      = (SELECT host FROM report_hosts"
+       "                         WHERE id = report_host_details.report_host)"
+       "             LIMIT 1)"
```

Comment:
I don't think LIMIT is ever considered inside EXISTS, so this line could be left out for neatness.

Reply:
I've addressed this and the following points in 0435471b341bf9b81a156b6835928c51ea5dfdb8.